mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/building/impl/PojoIndexedTypeManagerBuilder.java

```diff
@@ -140,6 +158,17 @@ void buildAndAddTo(PojoIndexedTypeManagerContainer.Builder typeManagersBuilder,
 		closed = true;
 	}
 
+	private Optional<BoundPojoModelPathPropertyNode<E, ?>> getEntityIdentifierPath() {
+		for ( Map.Entry<String, PojoPropertyAdditionalMetadata> entry
+				: typeAdditionalMetadata.getPropertiesAdditionalMetadata().entrySet() ) {
+			if ( entry.getValue().isEntityId() ) {
+				PojoPropertyModel<?> propertyModel = typeModel.getProperty( entry.getKey() );
+				return Optional.of( BoundPojoModelPath.root( typeModel ).property( propertyModel.getHandle() ) );
+			}
+		}
```

Comment:
We don't have a way to identify the id property when originally getting the information? It looks counter productive to iterate through the map of all the properties.

Reply:
It's not all properties, only those that have particular additional metadata; most of the properties won't be mentioned here.

About "identify the id property when originally getting the information", I'm not sure what "when originally getting the information" is referring to exactly...

I could move the name of the entity ID to the `PojoAdditionalMetadataEntityTypeNode`. I just pushed a fix-up commit with exactly that.

Alternatively, I guess we could use a callback to retrieve such information, similarly to what we do with `PojoTypeModel`, which would enable lazy retrieval of metadata in particular. Not sure it's worth the effort for now, especially since it's SPI that we could change in 6.1.